Make exec plugin more robust

awly opened the pull request:
1. rotate when cert is >1d old
2. rotate when cert expires in <1hr
3. if rotation fails, use old cert if still valid
4. if TPM attestation fails, use old bootstrap flow

awly commented:
/assign @mikedanese

mikedanese commented:
> if TPM attestation fails, use old bootstrap flow

I'm not sure I agree with this one. Shouldn't we decide this at VM creation time?

awly commented:
So, add another "legacy" mode (via flag) and set that in kubeconfig at node creation time?
That does sound way better, lemme rework this CL.

awly commented:
Reworked to always require TPM attestation to succeed.
Dropped `--bootstrap-kubeconfig` flag and loading `rest.Config` from `kube-env` metadata directly using new credential.

PTAL

mikedanese commented:
/lgtm

mikedanese commented:
/approve
